packages/material-ui/src/RadioGroup/RadioGroup.js

```diff
@@ -110,7 +110,7 @@ RadioGroup.propTypes = {
   /**
    * Value of the selected radio button.
    */
-  value: PropTypes.any,
+  value: PropTypes.string,
```

Comment:
What's wrong with any here?

Reply:
See 757384eb82df743fa20f04e9302552cb632240aa

Reply:
Correct. We would need to cast the values to a string in the comparison to support any:
```diff
-checked = radioGroup.value === props.value;
+checked = String(radioGroup.value) === String(props.value);
```

I believe it's the only change required.

Reply:
Have you looked at the test?

Reply:
I have. My proposed change would only support `string | number | boolean`. But as you said, it's a topic for another pull request. Is this important? I'm not sure.

Reply:
Pretty sure boolean would break easily and number is still dangerous.